infrastructure/src/main/java/org/corfudb/infrastructure/log/InMemoryStreamLog.java

```diff
@@ -84,8 +83,12 @@ public synchronized void prefixTrim(long address) {
     }
 
     @Override
-    public long getGlobalTail() {
-        return globalTail.get();
+    public synchronized Tails getTails() {
+        Map<UUID, Long> tails = new HashMap<>(logMetadata.getStreamTails().size());
+        for (Map.Entry<UUID, Long> entry : logMetadata.getStreamTails().entrySet()) {
+            tails.put(entry.getKey(), new Long(entry.getValue()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Long" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the boxing to "Long"; The argument is already of the same type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2153)